llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp

```diff
@@ -1229,7 +1229,13 @@ void initDefaultAMDKernelCodeT(AMDGPUMCKernelCodeT &KernelCode,
   KernelCode.amd_machine_version_minor = Version.Minor;
   KernelCode.amd_machine_version_stepping = Version.Stepping;
   KernelCode.kernel_code_entry_byte_offset = sizeof(amd_kernel_code_t);
-  KernelCode.wavefront_size = 6;
+  if (Version.Major >= 10 &&
```

Comment:
This probably doesn't need the version check, if you do it's hiding another bug somewhere. I guess this can be a separate change since it's already doing it

Reply:
I should add that this is explicitly checked for in the parser: https://github.com/llvm/llvm-project/blob/82c5d350d200ccc5365d40eac187b9ec967af727/llvm/lib/Target/AMDGPU/AsmParser/AMDGPUAsmParser.cpp#L5905-L5908

Reply:
Ah, I misunderstood the initial comment. I've removed the version check (although, do let me know if you'd like it to be a separate change after all).